serpent-moltres/fuel-column-sensitivity/postprocessing.py

```diff
@@ -0,0 +1,406 @@
+    It also prints the errors on the terminal.
+    Additionally, it obtains a weighted average of the error, using the
+    following weights:
+    - W_thermal = 0.5
+    - W_epithermal = 0.3
+    - W_fast = 0.2
+    However, this script doesn't run this function because I accidently
+    deleted the necessary files for reproducing the results.
+    We'll keep the function in case anyone wants to reproduce these
+    results in the future.
+    '''
+
+    lim26 = [4, 16, 26]  # from 26 to 3
+    lim15 = [2, 10, 15]  # from 15 to 3
+    lim15b = [2, 10, 15]  # from 15 to 3
+    lim15c = [2, 7, 15]  # from 15 to 3
+    lim15d = [2, 7, 15]  # from 15 to 3
+    lim15e = [2, 7, 15]  # from 15 to 3
+
+    e = np.zeros((5, 2, 3))
+    base = './accuracy15G/'
+    cumul = np.zeros((3, 5))
+
+    for lbp in ['noLBP', 'LBP']:
+        for temp in ['600', '1200']:
+
+            file1 = '3D-assembly-' + lbp + '-' + temp + '-26G_axial_0002.csv'
+            file2 = base + '3D-assembly-' + lbp + '-' + temp + \
+                '-15G_axial_0002.csv'
+            e[0] = np.array(calc_error(file1, file2, lim26, lim15))
+            file2 = base + '3D-assembly-' + lbp + '-' + temp + \
+                '-15Gb_axial_0002.csv'
+            e[1] = np.array(calc_error(file1, file2, lim26, lim15b))
+            file2 = base + '3D-assembly-' + lbp + '-' + temp + \
+                '-15Gc_axial_0002.csv'
+            e[2] = np.array(calc_error(file1, file2, lim26, lim15c))
+            file2 = base + '3D-assembly-' + lbp + '-' + temp + \
+                '-15Gd_axial_0002.csv'
+            e[3] = np.array(calc_error(file1, file2, lim26, lim15d))
+            file2 = base + '3D-assembly-' + lbp + '-' + temp + \
+                '-15Ge_axial_0002.csv'
+            e[4] = np.array(calc_error(file1, file2, lim26, lim15e))
+
+            print(f'{lbp} {temp}:')
+            cumul += np.round(e[:, 1, :].T*100, 1)
+            print(np.round(e[:, 1, :].T*100, 1))
+            xticks = ['a', 'b', 'c', 'd', 'e']
+            xlabel = 'Energy group structure'
+            plot_error_acc_study(e, 'accuracy-' + lbp + '-' + temp + '-er-15',
+                                 xticks, xlabel)
+
+    print('Cumulative error: ')
+    cumul = cumul/4
+    ave = np.zeros(5)
+    ave = 0.5*cumul[0, :] + 0.3*cumul[1, :] + 0.2*cumul[2, :]
+    print(ave)
+    return
+
+
+def noLBP_L2error(temp):
+    '''
+    This function plots the L2 norm relative error for the different
+    group structures for the no LBP cases.
+
+    Parameters:
+    -----------
+    temp: [float]
+        temperature in K: 600 or 1200K
+    '''
+
+    lim26 = [4, 16, 26]  # from 26 to 3
+    lim21 = [2, 12, 21]  # from 21 to 3
+    lim18 = [2, 11, 18]  # from 18 to 3
+    lim15 = [2, 10, 15]  # from 15 to 3
+    lim12 = [1, 7, 12]  # from 12 to 3
+    lim9 = [1, 5, 9]  # from 9 to 3
+    lim6 = [1, 3, 6]  # from 6 to 3
+    lim3 = [1, 2, 3]  # from 3 to 3
```

Comment:
should be a loop with loop logic and should avoid all of these hardcoded numbers.